### DISCUSSION TITLE
Make Graphite disabled by default

### DISCUSSION THREAD

**Discussion by ebraminio** [godotengine/godot-proposals#4855] [Engine Core]
Graphite is a supplemental font technology used inside Godot 4 text rendering stack but having it default enabled, shipping it along your builds for resource constraint environments like WebAssembly and Android apk feels a bit bloat and excessive, comparing it with other cross platform apps, it is enabled in Firefox but not in Unreal Engine and Chrome and Flutter for different considerations. It is also not available in Windows, Apple and Android default text stack.

I just want to make sure are you deliberately need it enabled by default, given low number of fonts using the technology as it shouldn't be widely in use in web as lack of support in the dominant OSes and browsers, and given font designers can use OpenType and specially AAT more or less for the same purpose so perhaps you can turn it into a default disabled dependency, before turning Godot 4 out of alpha.

Thanks!

cc: @bruvzg

**Comment by ebraminio** [comment]
Binary size difference of release_debug:

With graphite: 171,249,128 bytes
Without graphite: 171,057,560 bytes

It doesn't matter that much size wise as I used to think tbh, still.

**Reply by Calinou** [reply]
Did you run `strip` on the binary to remove debug symbols from it? This would make for a better binary size comparison.

**Comment by Calinou** [comment]
What features does Graphite provide? In other words, what do we lose by disabling Graphite at build-time?

**Reply by bruvzg** [reply]
> What features does Graphite provide? In other words, what do we lose by disabling Graphite at build-time?

It's adding support for a special format of "smart fonts", which essentially have custom shaping engine bytecode embedded into the font. It's mainly used for rare languages and scripts not supported by OpenType or for the fonts that designed to have non-standard contextual positioning. Graphite is quite small, I do not think it worth disabling by default, and it's already possible to disable it for the custom build (`graphite_enabled` option).
